diff --git a/libavcodec/libaribcaption.c b/libavcodec/libaribcaption.c
index 66d18b9d38..742186d72c 100644
--- a/libavcodec/libaribcaption.c
+++ b/libavcodec/libaribcaption.c
@@ -477,6 +477,7 @@ static int aribcaption_trans_bitmap_subtitle(ARIBCaptionContext *ctx)
             av_log(ctx, AV_LOG_WARNING, "CLUT overflow (%d).\n", ctx->clut_overflow);
     }
     sub->num_rects = rect_idx;
+    aribcc_render_result_cleanup(&ctx->render_result);
 
     return rect_idx;
 
@@ -492,6 +493,7 @@ fail:
         av_freep(&sub->rects);
     }
     sub->num_rects = 0;
+    aribcc_render_result_cleanup(&ctx->render_result);
 
     return ret;
 }
